katsdptelstate/telescope_state.py

```diff
@@ -681,14 +710,24 @@ def load_from_file(self, file):
         RdbParseError
             If the file could not be parsed (truncated / malformed / not RDB)
         """
-        keys_loaded = self._backend.load_from_file(file)
-        logger.info("Loading {} keys from {}".format(keys_loaded, file))
-        return keys_loaded
-
-    @classmethod
-    def join(cls, *names):
+        has_keys = bool(self.keys())
+        old_separator_bytes = self.SEPARATOR_BYTES
+        if has_keys:
+            logger.warning('load_from_file into a non-empty database has undefined behaviour')
```

Comment:
As discussed, we could probably just short circuit this whole thing and boot people out at this point. (and then add a sneaky expert mode flag back in to allow it)

Reply:
I think the first step is to add the "easy" mode where you give it a file or filename and you get back a telescope state object (with a memory backend). Then we can migrate code to use that, and declare load_from_file itself to be the "expert" mode (maybe under a different name to make that clearer). But I'm not sure I want to entangle that with this PR.

Reply:
Agreed - lets get this one done and then retrofit.